src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final char SEPARATOR = ',';
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Statistic statistic = readStatistic(fromFileName);
+        writeStatistic(statistic, toFileName);
```

Comment:
once again, your code totally violates SRP, goal of this task is to learn basics of SOLID principles, so kindly asking you to follow this approach or write your own approach, that will implement SRP principle.
```suggestion
        String dataFromFile = readFile(fromFileName);
        String report = createReport(dataFromFile);
        writeReportToFile(report, toFileName);
```

Reply:
in which lesson of the course was this explained?